core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java

```diff
@@ -871,8 +872,10 @@ public PlanOptimizers(
                         statsCalculator,
                         costCalculator,
                         ImmutableSet.<Rule<?>>builder()
-                                // Run RemoveEmptyDeleteRuleSet, RemoveEmptyUpdate and RemoveEmptyTableExecute after table scan is removed by PickTableLayout/AddExchanges
+                                // Run RemoveEmptyDeleteRuleSet, RemoveEmptyMergeWriterRuleSet, RemoveEmptyUpdate and RemoveEmptyTableExecute
```

Comment:
How about
```java
// run these after table scan is removed by AddExchanges
```
Looks like `PickTableLayout` has been gone for a long time...

Reply:
Changed.